tools/toolchains/iar.py

```diff
@@ -74,12 +76,15 @@ def __init__(self, target, notify=None, macros=None, build_profile=None,
         elif target.core == "Cortex-M7F":
             asm_flags_cmd += ["--fpu", "VFPv5_sp"]
             c_flags_cmd.append("--fpu=VFPv5_sp")
-        elif target.core == "Cortex-M23" or target.core == "Cortex-M33" or target.core == "Cortex-M33F":
-            self.flags["asm"] += ["--cmse"]
-            self.flags["common"] += ["--cmse"]
+        elif target.core.startswith("Cortex-M33F"):
+            asm_flags_cmd += ["--fpu", "VFPv5_sp"]
+            c_flags_cmd.append("--fpu=VFPv5_sp")
 
         # Create Secure library
-        if target.core == "Cortex-M23" or self.target.core == "Cortex-M33" or self.target.core == "Cortex-M33F":
+        if ((target.core.startswith("Cortex-M23") or target.core.startswith("Cortex-M33"))
```

Comment:
Could this be an architecture >=8 check (CORE_ARCH from tools.targets)?

(This is going to need revisiting at some point - shouldn't just assume that a v8 chip or M33 is going to be using separate non-secure and secure images).

Reply:
I am not sure if we have a way to check arch @theotherjimmy please suggest

Reply:
That would be the CORE_ARCH thing from tools.targets - was used in this commit I'm reverting: #9398

Reply:
What @kjbracey-arm said ^. That's the way to check arch version.

Reply:
Done 👍